### PR TITLE
web3-providers-ipc Test Coverage

### LINKED ISSUE #5745: web3-providers-ipc Test Coverage
Currently at 45.09%

Increase to 80%+

### PR DESCRIPTION
**Pull request by avkos**
## Description

Fixes #5745

image

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3-providers-ipc/test/unit/check_implementation.test.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import * as fs from 'fs';
+import * as net from 'net';
+import IpcProvider from '../../src/index';
+
+jest.mock('net');
+jest.mock('fs');
+
+// /* @ts-expect-error */
```
Comment (Spacesai1or) [inline packages/web3-providers-ipc/test/unit/check_implementation.test.ts]: Not sure if there's a reason for this syntax?

**Review by Spacesai1or — APPROVED** [review]

File: packages/web3-providers-ipc/test/unit/check_implementation.test.ts (hunk 1 above)
Comment (avkos) [reply packages/web3-providers-ipc/test/unit/check_implementation.test.ts]: fixed

**Review by Muhammad-Altabba — APPROVED** [review]

**Comment by nikoulai** [comment]
There is a related unit test failing, please check

**Review by jdevcs — APPROVED** [review]

**Review by jdevcs — COMMENTED** [review]
@avkos approved, but before merging check unit tests should be passing with this PR:

```
web3-providers-ipc: /home/runner/work/web3.js/web3.js/packages/web3-utils/lib/chunk_response_parser.js:38
web3-providers-ipc:                     throw new web3_errors_1.InvalidResponseError({
web3-providers-ipc:                     ^
web3-providers-ipc: InvalidResponseError: Returned error: Chunk timeout
``` 

if its due to this PR?

**Comment by avkos** [comment]
> @avkos approved, but before merging check unit tests should be passing with this PR:
> 
> ```
> web3-providers-ipc: /home/runner/work/web3.js/web3.js/packages/web3-utils/lib/chunk_response_parser.js:38
> web3-providers-ipc:                     throw new web3_errors_1.InvalidResponseError({
> web3-providers-ipc:                     ^
> web3-providers-ipc: InvalidResponseError: Returned error: Chunk timeout
> ```
> 
> if its due to this PR?

I don't know. It is not reproduced locally..

**Pull request merged**